packages/useValidators/index.js

```diff
@@ -0,0 +1,15 @@
+const { useState, createContext } = require('react');
+
+module.exports = function useValidators(validators) {
+  const [validationResults, setValidationResults] = useState([]);
+
+  function invokeValidators(value) {
+    const results = validators.map(function invokeValidator(validator) {
+      return validator(value);
```

Comment:
I'm a little mixed on the idea, but what if we just passed the arguments of `invokeValidators` into `validator`?

```js
// using es6 rest since it looks cleaner than using the arguments object
function invokeValidators(...args) {
  const results = validators.map(validator =>validator(...args));
  setValidationResults(results);
}
```

This approach opens up downstream `invokeValidators` usage and the validation rules we pass to `validators` to use more information than just `value` on a per-invocation basis.

Reply:
not sure I quite understand what you're saying. I don't see how this improves things. `value` is not a limiting factor because it can be anything.

Reply:
That's true. My original thought process was around passing functions that has more than just the single argument. While we can pass in an object that contains all those parameters there are times where having multiple parameters has a little more readability.

I think this is another case we can address with documentation. I don't think most people would think to use an object to contain those arguments despite most of us being used to the `props` object in react, so if anything else having an example would help.